src/main/src/main/java/org/geoserver/config/GeoServerDataDirectory.java

```diff
@@ -1202,7 +1203,7 @@ else if ( l.getResource() instanceof WMSLayerInfo ) {
     protected @Nonnull Style parsedStyleResources(StyleInfo s) throws IOException {
         final Resource styleResource = style(s);
         if ( styleResource.getType() == Type.UNDEFINED ){
-            throw new IOException( "No such resource: " + s.getFilename());
+            throw new FileNotFoundException( "No such resource: " + s.getFilename());
```

Comment:
Throwing a more specific exception here so we do not have to produce the full stack trace.